Cython/Compiler/ExprNodes.py

```diff
@@ -3871,7 +3871,7 @@ def analyse_as_cpp(self, env, setting):
         if self.exception_check:
             if not setting:
                 self.is_temp = True
-            if self.exception_value is None:
+            if self.exception_value is None or getattr(self.exception_value, "value", None) == "*":
```

Comment:
The repetition of this screams for a helper function, say, `needs_cpp_exception_conversion()`.
That also gives you enough space to spell out the cases in which `value` exists and is comparable, i.e. strings, I guess, instead of resorting to `getattr()`. Not sure if `value` can hold a byte string? (Which wouldn't compare equal to `"*"` in Py3.) Note that you could probably also use the `constant_value`.